### PR TITLE
🐛 Fixes MHC test failures

### PR DESCRIPTION
**Pull request by gab-satchi**
**What this PR does / why we need it**:
Fixes failures caused by #3186 

/assign @vincepri

### CODE REVIEW

File: controllers/machinehealthcheck_controller.go [hunk 1]
```diff
@@ -205,6 +206,7 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, logger log
 	for i, t := range targets {
 		m.Status.Targets[i] = t.Machine.Name
 	}
+	sort.Strings(m.Status.Targets)
```
Comment (detiber) [inline controllers/machinehealthcheck_controller.go:209]: Can't say I'm a huge fan of requiring sorting here just for the sake of the tests. Would it make sense to create a custom matcher to handle the test for equality instead?

File: controllers/machinehealthcheck_controller_test.go [hunk 2]
```diff
@@ -306,16 +307,19 @@ var _ = Describe("MachineHealthCheck", func() {
...
 					if err != nil {
-						return -1
+						return nil
 					}
-					return mhc.Status.CurrentHealthy
-				}).Should(Equal(int32(2)))
-				Expect(mhc.Status.ExpectedMachines).To(Equal(int32(2)))
-				Expect(mhc.Status.Targets).To(ConsistOf(targetMachines))
+					return &mhc.Status
+				}).Should(Equal(&clusterv1.MachineHealthCheckStatus{
```
Comment (fabriziopandini) [inline controllers/machinehealthcheck_controller_test.go:318]: > Should(Equal(&clusterv1.MachineHealthCheckStatus

This is going to fail when we https://github.com/kubernetes-sigs/cluster-api/pull/3226 will merge.
However the same PR will bring custom matchers for  `ExpectedMachines` and `CurrentHealthy`, so I'm +1 to get a custom matcher for `Targets` as well as suggested by @detiber

Comment (vincepri) [reply controllers/machinehealthcheck_controller_test.go:318]: Can we do the checks inside the expect call and return true or false for now?

Comment (gab-satchi) [reply controllers/machinehealthcheck_controller_test.go:316]: Added a custom matcher but if it's undesired, can switch to doing all the checks in the Eventually. Only issue I see with that is for test failures we'll get crappy messages like expected false to be true

Comment (vincepri) [reply controllers/machinehealthcheck_controller_test.go:316]: Looks good, thanks!

**Pull request merged**